### PR TITLE
[NETBEANS-54] Module Review httpserver

### PR DESCRIPTION
**Pull request by borxa**
Updated maven coordinates for the external binary.

Maven artifact version is 3.2.4 and file in source of netbeans was 3.2. The tomcat-webserver-3.2-license.txt is licensed to Apache 1.1. Need it change to 3.2.4 and Apache 2.0?

### CODE REVIEW

**Comment by vieiro** [comment]
Hi,

All tests should pass now, but I think the line
<runtime-relative-path>ext/webserver-3.2.4.jar</runtime-relative-path>
should look like 
<runtime-relative-path>ext/webserver.jar</runtime-relative-path>

The "runtime relative path" is the final destination path of the jar when the IDE/platform is built. If this is changed I think that any other modules referring to this jar may fail at runtime

File: httpserver/external/binaries-list [hunk 1]
```diff
@@ -14,4 +14,4 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-CD5B5996B46CB8D96C8F0F89A7A734B3C01F3DF7 tomcat-webserver-3.2.jar
+FCE6D146C0B03EF115B9AF05FF2976F15308998C tomcat:webserver:3.2.4
```
Comment (matthiasblaesing) [inline httpserver/external/binaries-list]: Did you check in any way, that this still works? What is the difference between the jars?

File: httpserver/external/tomcat-webserver-3.2-license.txt [hunk 2]
```diff
@@ -1,58 +0,0 @@
-Name: Apache Tomcat
```
Comment (matthiasblaesing) [inline httpserver/external/tomcat-webserver-3.2-license.txt]: Please add a `webserver-3.2.4-license.txt` file. You can start from this file and update version and license information.

File: httpserver/nbproject/project.properties [hunk 3]
```diff
@@ -15,10 +15,11 @@
 # specific language governing permissions and limitations
 # under the License.
 
+file.reference.webserver-3.2.4.jar=release/modules/ext/webserver-3.2.4.jar
```
Comment (matthiasblaesing) [inline httpserver/nbproject/project.properties]: Why was this added?

File: httpserver/nbproject/project.xml [hunk 4]
```diff
@@ -153,7 +153,7 @@
             <public-packages/>
             <class-path-extension>
                 <runtime-relative-path>ext/webserver.jar</runtime-relative-path>
-                <binary-origin>external/tomcat-webserver-3.2.jar</binary-origin>
+                <binary-origin>release/modules/ext/webserver-3.2.4.jar</binary-origin>
```
Comment (matthiasblaesing) [inline httpserver/nbproject/project.xml]: This should be `external/webserver-3.24.jar` (the relative path to the downloaded binary)

File: httpserver/external/binaries-list (hunk 1 above)
Comment (borxa) [reply httpserver/external/binaries-list]: Version 3.2.4 is unique version in maven public https://search.maven.org/#search%7Cga%7C1%7Ca%3A%22webserver%22

It's similiar to 3.2, adds a new package org.apache.tomcat.util.compat

Build code, run with ant tryme, update IDE, install Java EE plugins, add Liferay Tomcat server, config it, run it, stop it, without problems like a my Netbeans 8.2 installed version.

File: httpserver/nbproject/project.properties (hunk 3 above)
Comment (borxa) [reply httpserver/nbproject/project.properties]: Solved

File: httpserver/nbproject/project.xml (hunk 4 above)
Comment (borxa) [reply httpserver/nbproject/project.xml]: Solved

File: httpserver/external/tomcat-webserver-3.2-license.txt (hunk 2 above)
Comment (borxa) [reply httpserver/external/tomcat-webserver-3.2-license.txt]: I don't know why ignore it with /*/external/*/** line in .gitignore. Commit it.

**Comment by matthiasblaesing** [comment]
Thank you very much. From my perspective the testing part was important. I squashed your commits together and pushed them as one commit.
